### PR TITLE
ndc-spec 0.1.2

### PR DESCRIPTION
**Pull request by soupi**
### What

Support ndc-spec v0.1.2.
This includes changing the type representations of some types.

### How

The type representation has changed. We try to keep this change backwards compatible by mapping deprecated variants (Number, Integer) to new variants, and update types to more specific reps that didn't exist before.

We also have a problem with representing int8 now. ndc-spec v0.1.2 defines Int64 to be represented as a json string, which is not what we do. So we either have to lie (say it's an Int32 when it isn't) or break compatibility and cast int8 to string.
In this PR I did the former, and I'll introduce the latter change next.

### CODE REVIEW

File: crates/connectors/ndc-postgres/src/schema.rs
Comment (soupi) [inline crates/connectors/ndc-postgres/src/schema.rs:623]: Stop gap until we remove these.

**Review by SamirTalwar — APPROVED** [review]

**Comment by SamirTalwar** [comment]
Dismissing my review because I just realized that the ndc-sdk commit points to a branch. Let's wait until that's merged and update the commit.

File: crates/connectors/ndc-postgres/src/schema.rs
Comment (danieljharvey) [inline crates/connectors/ndc-postgres/src/schema.rs:619]: I'm not sure we should be changing the outputted type representations like this without the user providing us an opt-in config flag to do so.

Comment (soupi) [reply crates/connectors/ndc-postgres/src/schema.rs:619]: This part is just translating our own TypeRepresentation type to ndc-spec TypeRepresentation type. Could you please elaborate on what do you mean? I'm not sure I understand.

Comment (danieljharvey) [reply crates/connectors/ndc-postgres/src/schema.rs:619]: I was under the impression that our types should return the same `ndc-spec` `TypeRepresentation` as they did already before this change. Where this is not possible, because of removed types, we should return the more general `json`. Otherwise we're potentially changing behaviour without the user opting into it.

File: static/cockroach/v3-chinook-ndc-metadata/configuration.json
Comment (soupi) [inline static/cockroach/v3-chinook-ndc-metadata/configuration.json:2811]: These are the changes to the configuration. Number and String are deprecated in ndc-spec. We will still accept these for backwards compatibility reasons, but we will output the new representations. @danieljharvey

File: crates/tests/databases-tests/src/postgres/snapshots/databases_tests__postgres__schema_tests__schema_test__get_schema.snap
Comment (soupi) [inline crates/tests/databases-tests/src/postgres/snapshots/databases_tests__postgres__schema_tests__schema_test__get_schema.snap:755]: These are the changes to the ndc schema.

File: crates/connectors/ndc-postgres/src/schema.rs
Comment (danieljharvey) [reply crates/connectors/ndc-postgres/src/schema.rs:619]: What will a version of engine that does not know about these types do when it receives them?

Comment (soupi) [reply crates/connectors/ndc-postgres/src/schema.rs:619]: This is an ndc-schema concept that can be used by cli/lsp to generate engine objects, so I don't think this will be used directly in the engine. The engine already uses v0.1.2 of ndc-spec though: https://github.com/hasura/v3-engine/pull/443/files#diff-5e05c14aa4760c15d75bc66970581abe17a82eef4770a0cd9861bd40ce65934bR27

Comment (danieljharvey) [reply crates/connectors/ndc-postgres/src/schema.rs:619]: Ahh OK, makes sense. This all seems sensible then, thank you for indulging my questions.

**Review by danieljharvey — APPROVED** [review]

File: Cargo.toml
Comment (SamirTalwar) [inline Cargo.toml:51]: I would prefer to use the full SHA here because Cargo thinks two dependencies pointing to the _same_ revision but using a _different_ prefix length are actually different dependencies. This causes serious issues when using ndc-postgres as a dependency as it brings in ndc-sdk with the short ref, so everything else needs to use the exact same prefix length.

Comment (soupi) [reply Cargo.toml:51]: Good to know. Will fix.

**Pull request merged**
